packages/did/src/Did.chain.ts

```diff
@@ -558,10 +612,15 @@ export async function generateDidAuthenticatedTx({
 
 // ### Chain utils
 export function encodeDidSignature(
-  key: Pick<DidKey, 'type'>,
+  key: Pick<ChainDidKey, 'type'>,
   signature: Pick<DidSignature, 'signature'>
 ): SignatureEnum {
-  const alg = getSignatureAlgForKeyType(key.type)
+  if (!Object.keys(VerificationKeyType).some((kt) => kt === key.type)) {
```

Comment:
you have a helper function for that in the utils, right?

Reply:
The helper function works for `NewDidKey` and `DidKey` types, which are DID keys. This one here deals with chain keys, whose type is a string, that is why here it iterates over the keys while in the util function it iterates over the values.